### DISCUSSION TITLE
Strength bits for integrity-only cipher suites

### DISCUSSION THREAD

**Discussion by rajeev-0** [openssl/openssl#22987] [General Discussion]
I am implementing integrity-only cipher suites (RFC9150) in OpenSSL 3.3 (https://github.com/openssl/openssl/pull/22903) where I need to provide the strength bits value for TLS_SHA256_SHA256 (uses HMAC-SHA256) & TLS_SHA384_SHA384 (uses HMAC-SHA384).

I am not sure how these strength bits values are derived. For integrity-only cipher, there is no encryption and only using HMAC.
I need help understanding the values which could be set for `strength_bits` for TLS_SHA256_SHA256 and TLS_SHA384_SHA384?

**Comment by t8m** [comment]
This is debatable - one option would be to set it based on the hash function strength. The other option would be to set it to zero. I am not sure what would be the implications of these choices. @mattcaswell might have an opinion which would be more appropriate.

**Comment by mattcaswell** [comment]
All existing integrity-only cipher suites (i.e. the TLSv1.2 ones) set 0 for the strength bits - so i think we should do the same for the TLSv1.3 ones.
